meta/applicator.json

```diff
@@ -2,7 +2,8 @@
     "$schema": "https://json-schema.org/draft/next/schema",
     "$id": "https://json-schema.org/draft/next/meta/applicator",
     "$vocabulary": {
-        "https://json-schema.org/draft/next/vocab/applicator": true
+        "https://json-schema.org/draft/next/vocab/applicator": true,
+        "https://json-schema.org/draft/next/vocab/vocabulary": false
```

Comment:
This says that schemas using `https://json-schema.org/draft/next/meta/applicator` as a meta-schema are able to use the vocabulary vocabulary, which I don't think is what you mean.  You need this meta-schema's meta-schema (currently declared as `https://json-schema.org/draft/next/schema`) to include the vocabulary vocabulary in `$vocabulary` (buffalo buffalo buffalo buffalo buffalo...).  In which case it probably would not be the default meta-schema because most schemas won't need to (and shouldn't) use the vocabulary vocabulary.

Reply:
> You need this meta-schema's meta-schema (currently declared as `https://json-schema.org/draft/next/schema`) to include the vocabulary vocabulary in $vocabulary

But if that's just `https://json-schema.org/draft/next/schema`, it doesn't solve the problem you're describing.

What I want is for meta-schemas that describe vocabularies to be able to use these new keywords.  Does that mean that we need a dedicated vocabulary-describing meta-schema?  Then have (e.g.) applicator meta-schema reference that one in `$schema`?

So

```json
{
  "$schema": "https://json-schema.org/draft/next/schema",
  "$id": "https://json-schema.org/draft/next/meta/vocabulary",
  "$vocabulary": {
    "https://json-schema.org/draft/next/vocab/core": true,
    "https://json-schema.org/draft/next/vocab/applicator": true,
    "https://json-schema.org/draft/next/vocab/unevaluated": true,
    "https://json-schema.org/draft/next/vocab/validation": true,
    "https://json-schema.org/draft/next/vocab/meta-data": true,
    "https://json-schema.org/draft/next/vocab/format-annotation": true,
    "https://json-schema.org/draft/next/vocab/content": true,
    "https://json-schema.org/draft/next/vocab/vocabulary": true
  },
  "$ref": "https://json-schema.org/draft/next/schema",
  ... // new vocab-vocab properties
}
```

Then `https://json-schema.org/draft/next/meta/applicator` and friends all have `$schema: https://json-schema.org/draft/next/meta/vocabulary`?

I think this also provides a pre-packaged meta-schema for other custom meta-schemas to use.

Reply:
But if it has `$schema: https://json-schema.org/draft/next/schema` then it can't itself use the keywords it defines.  So does it need to be its own meta-schema?  That line of logic leads us to [Keyword for identifying bootstrapping rules #217](https://github.com/orgs/json-schema-org/discussions/217).

Reply:
`$vocabulary` is (effectively) an annotation.  So when you use it in a meta-schema, it annotates the (non-meta-)schema and says "this schema can use this vocabulary".

If this were not the case, then every non-meta-schema would have to declare `$vocabulary`, which would be a mess.

So yes, if you want other vocab meta-schemas to use the vocabulary vocabulary (VV), then the VV needs to be in *their* meta-schema's `$vocabulary`.  Since we would not want to put the VV in the default meta-schema, yes that means they would need a different meta-schema.

Reply:
@gregsdennis 

> This data is still mostly informational. I don't expect an implementation to be able to figure out what to do with an applicator or assertion without some additional coding, just like today. I don't think we'll ever get to the point where an implementation can "just know" what to do with a new keyword that's not a pure annotation.
>
> If this isn't what you're talking about, perhaps you can elaborate on what you mean by "take action." What action would you expect from an implementation?

At minimum, we need to be able to use any vocabulary description to automatically determine what keywords are part of that vocabulary, which will let us distinguish known keywords from optional (and not directly-supported) vocabularies from completely unknown keywords.  Otherwise, it's just JSON-formatted documentation, and that doesn't seem useful to me.

Reply:
> So yes, if you want other vocab meta-schemas to use the vocabulary vocabulary (VV), then the VV needs to be in their meta-schema's $vocabulary. Since we would not want to put the VV in the default meta-schema, yes that means they would need a different meta-schema.

I think this means that you agree with my [commented approach](https://github.com/json-schema-org/json-schema-spec/pull/1257#discussion_r981977106).

Reply:
@gregsdennis yes, I think so (I think I got confused by the next comment after that)